crates/bevy_transform/src/systems.rs

```diff
@@ -89,100 +131,404 @@ pub fn propagate_transforms(
+        let task_pool = ComputeTaskPool::get_or_init(TaskPool::default);
+        task_pool.scope(|s| {
+            (0..task_pool.thread_num())
+                .for_each(|_| s.spawn(async { propagation_worker(&queue, &nodes) }));
+        });
+    }
+
+    /// A parallel worker that will consume processed parent entities from the queue, and push
+    /// children to the queue once it has propagated their [`GlobalTransform`].
+    #[inline]
+    fn propagation_worker(queue: &WorkQueue, nodes: &NodeQuery) {
+        #[cfg(feature = "std")]
+        let _span = bevy_log::info_span!("transform propagation worker").entered();
+
+        let mut outbox = queue.local_queue.borrow_local_mut();
+        loop {
+            // Try to acquire a lock on the work queue in a tight loop. Profiling shows this is much
+            // more efficient than relying on `.lock()`, which causes gaps to form between tasks.
+            let Ok(rx) = queue.receiver.try_lock() else {
+                continue;
+            };
+            // If the queue is empty and no other threads are busy processing work, we can conclude
+            // there is no more work to do, and end the task by exiting the loop.
+            let Some(mut tasks) = rx.try_iter().next() else {
+                if queue.busy_threads.load(Ordering::Relaxed) == 0 {
+                    break; // All work is complete, kill the worker
+                }
+                continue; // No work to do now, but another thread is busy creating more work.
+            };
+            if tasks.is_empty() {
+                continue; // This shouldn't happen, but if it does, we might as well stop early.
+            }
+
+            // At this point, we know there is work to do, so we increment the busy thread counter,
+            // and drop the mutex guard *after* we have incremented the counter. This ensures that
+            // if another thread is able to acquire a lock, the busy thread counter will already be
+            // incremented.
+            queue.busy_threads.fetch_add(1, Ordering::Relaxed);
+            drop(rx); // Important: drop after atomic and before work starts.
+
+            for parent in tasks.drain(..) {
+                // SAFETY: each task pushed to the worker queue represents an unprocessed subtree of
+                // the hierarchy, guaranteeing unique access.
+                #[expect(unsafe_code, reason = "Mutating disjoint entities in parallel")]
+                unsafe {
+                    let (_, (_, p_global_transform), (p_children, _)) =
+                        nodes.get_unchecked(parent).unwrap();
+                    propagate_descendants_unchecked(
+                        parent,
+                        p_global_transform,
+                        p_children,
+                        nodes,
+                        &mut outbox,
+                        queue,
+                        10_000,
```

Comment:
Empirically selected? :)

Reply:
You could call it that. :)
It's big enough to be larger than any test or scene we have, but small enough that it won't hang the application. I'm actually looking at tweaking it to improve work sharing.